crates/exex/src/hints.rs

```diff
@@ -112,3 +119,188 @@ pub fn add_segment_hint() -> Hint {
+            MaybeRelocatable::from(Felt252::from_bytes_be_slice(&header.receipts_root.0[0..16])),
+            MaybeRelocatable::from(Felt252::from_bytes_be_slice(&header.state_root.0[16..])),
+            MaybeRelocatable::from(Felt252::from_bytes_be_slice(&header.state_root.0[0..16])),
+            MaybeRelocatable::from(Felt252::from(header.timestamp)),
+            MaybeRelocatable::from(Felt252::from_bytes_be_slice(&header.transactions_root.0[16..])),
+            MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.transactions_root.0[0..16],
+            )),
+            MaybeRelocatable::from(Felt252::from_bytes_be_slice(&header.ommers_hash.0[16..])),
+            MaybeRelocatable::from(Felt252::from_bytes_be_slice(&header.ommers_hash.0[0..16])),
+            MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.withdrawals_root.unwrap_or_default().0[16..],
+            )),
+            MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.withdrawals_root.unwrap_or_default().0[0..16],
+            )),
+        ]
+    }
+
+    fn to_cairo_vm_block_body(&self) -> Vec<MaybeRelocatable> {
+        // The first element is the length of the body (number of transactions)
+        let mut block_body = vec![MaybeRelocatable::from(Felt252::from(self.body.len()))];
+
+        self.body.iter().for_each(|transaction| {
+            // RLP encode the transaction
+            let mut buf = Vec::new();
+            transaction.encode_without_signature(&mut buf);
+
+            block_body.extend([
+                // The length of the RLP encoded transaction
+                MaybeRelocatable::from(Felt252::from(buf.len())),
+                // The RLP encoded transaction
+                MaybeRelocatable::from(Felt252::from_bytes_be_slice(&buf)),
+                // The length of the signature payload
+                MaybeRelocatable::from(Felt252::from(transaction.signature.payload_len())),
+                // The signature payload
+                MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                    &transaction.signature.to_bytes(),
+                )),
+            ]);
+        });
+
+        block_body
+    }
+}
+
+pub fn block_hint(block: SealedBlock) -> Hint {
+    Hint::new(
+        String::from("block"),
+        move |vm: &mut VirtualMachine,
+              _exec_scopes: &mut ExecutionScopes,
+              ids_data: &HashMap<String, HintReference>,
+              ap_tracking: &ApTracking,
+              _constants: &HashMap<String, Felt252>|
+              -> HintExecutionResult {
+            // We retrieve the `model.Block*` pointer from the `ids_data` hashmap.
+            // This pointer is used to store the block-related values in the VM.
+            let block_ptr = get_ptr_from_var_name("block", vm, ids_data, ap_tracking)?;
+
+            vm.load_data(block_ptr, &block.to_cairo_vm_block()).map_err(HintError::Memory)?;
```

Comment:
here I'd like to use a higher level method consuming a properly defined data type https://github.com/lambdaclass/cairo-vm/blob/main/vm/src/vm/vm_memory/memory_segments.rs#L132-L146

Reply:
Ok so I used simply:

```rust
// This call will first add a new memory segment to the VM (the base)
// Then we load the block into the VM starting from the base
vm.gen_arg(&block.to_cairo_vm_block())?;
```

where the argument will be downcasted automatically to a vec of MaybeRelocatable, it includes the generation of the new segment and the load of the data.